ga4gh/datamodel/variants.py

```diff
@@ -787,9 +791,19 @@ def buildMetadata(
         return ret
 
     def isAnnotated(self, path):
+        # assumes that all files in the directory look like the first
+        return self.hasAnnField(path) or self.hasConsequenceField(path)
+
+    def hasAnnField(self, path):
+        return 'ANN' in self._getHeaderItems(path)
+
+    def hasConsequenceField(self, path):
+        return 'CSQ' in self._getHeaderItems(path)
+
+    def _getHeaderItems(self, path):
         pysamreader = self.openFile(
             glob.glob(os.path.join(path, "*.vcf.gz"))[0])
-        return 'ANN' in [x[0] for x in pysamreader.header.info.items()]
```

Comment:
We shouldn't need to parse the header each time to answer these (which we only use in the constructor anyway?). VCF headers and big and complicated so this will be slow. We should try to read the header information once and store the information sufficient to answer the questions we need to ask during the lifetime of the object as instance variables.